siriuspy/siriuspy/devices/orbit_interlock.py

```diff
@@ -101,25 +227,1189 @@ class BPMInterlock(_Device):
+        for dev in self.devices:
+            dev.cmd_reset_gen()
+        return True
+
+    @property
+    def gen_inst(self):
+        """General instantaneous interlocks.
+
+        Returns:
+            gen_inst (numpy.ndarray, 160):
+                general instantaneous interlock for each BPM.
+        """
+        return _np.array([b.gen_inst for b in self._devices])
+
+    @property
+    def gen_latch(self):
+        """General latch interlocks.
+
+        Returns:
+            gen_latch (numpy.ndarray, 160):
+                general latch interlock for each BPM.
+        """
+        return _np.array([b.gen_latch for b in self._devices])
+
+    # --- minimum sum threshold ---
+
+    def cmd_minsumthres_enable(self, timeout=TIMEOUT):
+        """Enable all BPM minimum sum threshold."""
+        for dev in self.devices:
+            dev.minsumthres_enable = 1
+        return self._wait_devices_propty(
+            self.devices, 'IntlkMinSumEn-Sts', 1, timeout=timeout)
+
+    def cmd_minsumthres_disable(self, timeout=TIMEOUT):
+        """Disable all BPM minimum sum threshold."""
+        for dev in self.devices:
+            dev.minsumthres_enable = 0
+        return self._wait_devices_propty(
+            self.devices, 'IntlkMinSumEn-Sts', 0, timeout=timeout)
+
+    @property
+    def minsumthres(self):
+        """Minimum sum thresholds.
+
+        Returns:
+            thres (numpy.ndarray, 160): min.sum threshold for each BPM.
+        """
+        return _np.array([b.minsumthres for b in self._devices])
+
+    @minsumthres.setter
+    def minsumthres(self, value):
+        if isinstance(value, (int, float)):
+            value = [value, ] * len(self.BPM_NAMES)
+        for idx, dev in enumerate(self.devices):
+            dev.minsumthres = value[idx]
+
+    # --- translation interlock ---
+
+    def cmd_trans_enable(self, timeout=TIMEOUT):
+        """Enable all BPM translation interlock."""
+        for dev in self.devices:
+            dev.trans_enable = 1
+        return self._wait_devices_propty(
+            self.devices, 'IntlkTransEn-Sts', 1, timeout=timeout)
+
+    def cmd_trans_disable(self, timeout=TIMEOUT):
+        """Disable all BPM translation interlock."""
+        for dev in self.devices:
+            dev.trans_enable = 0
+        return self._wait_devices_propty(
+            self.devices, 'IntlkTransEn-Sts', 0, timeout=timeout)
+
+    def cmd_reset_trans(self):
+        """Reset all BPM translation interlock."""
+        for dev in self.devices:
+            dev.cmd_reset_trans()
+        return True
+
+    @property
+    def trans_thresminx(self):
+        """Minimum x translation thresholds.
+
+        Returns:
+            thres (numpy.ndarray, 160):
+                min. x translation threshold for each BPM.
+        """
+        return _np.array([b.trans_thresminx for b in self._devices])
+
+    @trans_thresminx.setter
+    def trans_thresminx(self, value):
+        if isinstance(value, (int, float)):
```

Comment:
humm, this is not generic enough, maybe include `_np.int` and `_np.float` to the tuple.
It would also be nice to check the length of `value` before start setting the properties in case the user already provide an iterable.

The same is valid for the other properties.

Reply:
done!